### PR TITLE
Add SDMMC and example

### PR DESCRIPTION
**Pull request by richardeoin**
Uses [sdio-host](https://crates.io/crates/sdio-host).

Requires `sdmmc` feature gate.

### CODE REVIEW

File: src/sdmmc.rs [hunk 1]
```diff
@@ -0,0 +1,1185 @@
...
+                    direction: Dir,
+                ) {
+                    // Block Size up to 2^14 bytes
+                    assert!(block_size <= 14);
+
+                    // Block Size must be greater than 0 ( != 1 byte) in DDR mode
+                    let ddr = self.sdmmc.clkcr.read().ddr().bit_is_set();
+                    assert!(
+                        !ddr || block_size != 0,
+                        "Block size must be >= 1, or >= 2 is DDR mode"
```
Comment (hargoniX) [inline src/sdmmc.rs:612]: **in** DDR mode

File: src/sdmmc.rs [hunk 2]
```diff
@@ -0,0 +1,1185 @@
...
+        D3: [PC11<Alternate<AF12>>]
+        D4: [PB8<Alternate<AF12>>]
+        D5: [PB9<Alternate<AF12>>]
+        D6: [PC6<Alternate<AF12>>]
+        D7: [PC7<Alternate<AF12>>]
+        CKIN: [PB8<Alternate<AF7>>]
+        CDIR: [PB9<Alternate<AF7>>]
+        D0DIR: [PC6<Alternate<AF8>>]
+        D123DIR: [PC7<Alternate<AF8>>]
+        SDMMC2:
```
[the remark this answers is not in the document]
Comment (richardeoin) [reply src/sdmmc.rs]: Fixed

**Comment by richardeoin** [comment]
@hargoniX Thanks, I fixed up those points. The error messages in the `assert!`s are helpful, I've added some more

**Pull request merged**